tls/s2n_resume.c

```diff
@@ -86,6 +86,16 @@ static S2N_RESULT s2n_tls13_serialize_resumption_state(struct s2n_connection *co
     RESULT_GUARD_POSIX(s2n_stuffer_write_uint8(out, ticket_fields->session_secret.size));
     RESULT_GUARD_POSIX(s2n_stuffer_write_bytes(out, ticket_fields->session_secret.data, ticket_fields->session_secret.size));
 
+    uint32_t server_max_early_data = 0;
+    RESULT_GUARD(s2n_early_data_get_server_max_size(conn, &server_max_early_data));
+    RESULT_GUARD_POSIX(s2n_stuffer_write_uint32(out, server_max_early_data));
+    if (server_max_early_data > 0) {
+        RESULT_GUARD_POSIX(s2n_stuffer_write_uint8(out, strlen(conn->application_protocol)));
```

Comment:
NitPick: We should save the results of calling `strlen` to avoid needing to calculate it twice. (I mean, it will be super fast because it can only be up to 255 characters long, but still...)